src/Pact/Types/Util.hs

```diff
@@ -68,21 +71,36 @@ lensyParseJSON n = genericParseJSON (lensyOptions n)
 newtype Hash = Hash { unHash :: ByteString }
   deriving (Eq, Ord, Generic, Hashable)
 instance Show Hash where
-  show (Hash h) = show $ B16.encode h
+  show (Hash h) = show $ b64URL_encode h
 instance Pretty Hash where
   pretty = pretty . asString
-instance AsString Hash where asString (Hash h) = decodeUtf8 (B16.encode h)
+instance AsString Hash where asString (Hash h) = decodeUtf8 (b64URL_encode h)
 instance NFData Hash
 
+-- | TODO make PR for https://github.com/haskell/base64-bytestring/issues/18
+b64URL_encode :: ByteString -> ByteString
+b64URL_encode = B.init . B64URL.encode
```

Comment:
Add a comment that is function only works for 256 bit values.

Reply:
I decided instead to adapt your padding code from chainweb